<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>bc_upper</class>
 <widget class="QMainWindow" name="bc_upper">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1199</width>
    <height>754</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>bc_upper</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,5">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3" stretch="1">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,5,2">
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>端口连接</string>
            </property>
            <widget class="QWidget" name="gridLayoutWidget">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>20</y>
               <width>181</width>
               <height>63</height>
              </rect>
             </property>
             <layout class="QGridLayout" name="gridLayout" rowstretch="0,0,0" columnstretch="1,0">
              <property name="leftMargin">
               <number>1</number>
              </property>
              <property name="topMargin">
               <number>1</number>
              </property>
              <property name="rightMargin">
               <number>1</number>
              </property>
              <property name="bottomMargin">
               <number>1</number>
              </property>
              <item row="0" column="1">
               <widget class="QComboBox" name="cb_port"/>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>端口：</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <spacer name="verticalSpacer_7">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="2" column="1">
               <widget class="QPushButton" name="push_open">
                <property name="text">
                 <string>打开</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QPushButton" name="push_scan">
                <property name="text">
                 <string>扫描</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>参数设置</string>
            </property>
            <widget class="QWidget" name="gridLayoutWidget_2">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>17</y>
               <width>181</width>
               <height>421</height>
              </rect>
             </property>
             <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,2" columnminimumwidth="0,0">
              <item row="13" column="0">
               <widget class="QLabel" name="lb_speed_p">
                <property name="text">
                 <string>速度P</string>
                </property>
               </widget>
              </item>
              <item row="17" column="1">
               <widget class="QPushButton" name="push_speed_i">
                <property name="text">
                 <string>设置</string>
                </property>
               </widget>
              </item>
              <item row="11" column="0">
               <widget class="QLineEdit" name="le_balance_d">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item row="5" column="1">
               <widget class="QPushButton" name="push_balance_p">
                <property name="text">
                 <string>设置</string>
                </property>
               </widget>
              </item>
              <item row="11" column="1">
               <widget class="QPushButton" name="push_balance_d">
                <property name="text">
                 <string>设置</string>
                </property>
               </widget>
              </item>
              <item row="22" column="0">
               <widget class="QLabel" name="lb_speed">
                <property name="text">
                 <string>运行速度：</string>
                </property>
               </widget>
              </item>
              <item row="6" column="0">
               <spacer name="verticalSpacer">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="17" column="0">
               <widget class="QLineEdit" name="le_speed_i">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item row="7" column="0">
               <widget class="QLabel" name="lb_balance_i">
                <property name="text">
                 <string>平衡I:</string>
                </property>
               </widget>
              </item>
              <item row="16" column="0">
               <widget class="QLabel" name="lb_speed_i">
                <property name="text">
                 <string>速度I:</string>
                </property>
               </widget>
              </item>
              <item row="21" column="0">
               <spacer name="verticalSpacer_6">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="15" column="0">
               <spacer name="verticalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="20" column="0">
               <widget class="QLineEdit" name="le_speed_d">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item row="9" column="0">
               <spacer name="verticalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="18" column="0">
               <spacer name="verticalSpacer_5">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="20" column="1">
               <widget class="QPushButton" name="push_speed_d">
                <property name="text">
                 <string>设置</string>
                </property>
               </widget>
              </item>
              <item row="23" column="1">
               <widget class="QPushButton" name="push_speed">
                <property name="text">
                 <string>设置</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="lb_balance_p">
                <property name="text">
                 <string>平衡P:</string>
                </property>
               </widget>
              </item>
              <item row="10" column="0">
               <widget class="QLabel" name="lb_balance_d">
                <property name="text">
                 <string>平衡D:</string>
                </property>
               </widget>
              </item>
              <item row="14" column="0">
               <widget class="QLineEdit" name="le_speed_p">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item row="5" column="0">
               <widget class="QLineEdit" name="le_balance_p">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item row="8" column="1">
               <widget class="QPushButton" name="push_balance_i">
                <property name="text">
                 <string>设置</string>
                </property>
               </widget>
              </item>
              <item row="14" column="1">
               <widget class="QPushButton" name="push_speed_p">
                <property name="text">
                 <string>设置</string>
                </property>
               </widget>
              </item>
              <item row="23" column="0">
               <widget class="QLineEdit" name="le_speed">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item row="19" column="0">
               <widget class="QLabel" name="lb_speed_d">
                <property name="text">
                 <string>速度D:</string>
                </property>
               </widget>
              </item>
              <item row="8" column="0">
               <widget class="QLineEdit" name="le_balance_i">
                <property name="text">
                 <string>0</string>
                </property>
               </widget>
              </item>
              <item row="12" column="0">
               <spacer name="verticalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_3">
            <property name="title">
             <string>方向控制</string>
            </property>
            <widget class="QWidget" name="gridLayoutWidget_3">
             <property name="geometry">
              <rect>
               <x>10</x>
               <y>20</y>
               <width>181</width>
               <height>161</height>
              </rect>
             </property>
             <layout class="QGridLayout" name="gridLayout_3" rowstretch="1,1,1" columnstretch="1,1,1">
              <item row="0" column="0">
               <widget class="QPushButton" name="push_left_up">
                <property name="text">
                 <string>左上</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QPushButton" name="push_up">
                <property name="text">
                 <string>上</string>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QPushButton" name="push_right_up">
                <property name="text">
                 <string>右上</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QPushButton" name="push_left">
                <property name="text">
                 <string>左</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QPushButton" name="push_stop">
                <property name="text">
                 <string>O</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QPushButton" name="push_left_down">
                <property name="text">
                 <string>左下</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QPushButton" name="push_down">
                <property name="text">
                 <string>下</string>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QPushButton" name="push_right">
                <property name="text">
                 <string>右</string>
                </property>
               </widget>
              </item>
              <item row="2" column="2">
               <widget class="QPushButton" name="push_right_down">
                <property name="text">
                 <string>右下</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
